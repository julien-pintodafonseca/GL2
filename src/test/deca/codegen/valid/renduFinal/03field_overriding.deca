// Description:
//      Définition d'un attribut pour une classe A qui étend une classe B
//      => Cas spécifique où l'attribut de la classe A a le même nom que l'attribut de la classe B (overriding)
//
// Resultats:
//      xA=3
//      xB=2
//      xBA=2

class B {
    int x=2;
}

class A extends B {
    int x=3;
}

{
    A a = new A();
    B b = new B();
    B ba = new A();
    println("xA=", a.x);
    println("xB=", b.x);
    println("xBA=", ba.x);
}