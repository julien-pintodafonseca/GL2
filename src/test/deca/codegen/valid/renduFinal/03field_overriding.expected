xA=3
xB=2
xBA=2
